stream_alert_cli/athena/handler.py

```diff
@@ -0,0 +1,312 @@
+        if not success:
+            LOGGER_CLI.error('Unable to drop the %s table', table)
+        else:
+            LOGGER_CLI.info('Dropped %s', table)
+
+
+def _add_to_athena_schema(log_schema, athena_schema, root_key=''):
+    """Add sanitized schemas to the Athena table schema
+
+    Args:
+        log_schema (dict): The related StreamAlert log schema
+        athena_schema (dict): The Athena table schema to add to
+        root_key (str): The hierarchy where the schema should be added
+    """
+    # Setup the root_key dict
+    if root_key and not athena_schema.get(root_key):
+        athena_schema[root_key] = {}
+
+    for key_name, key_type in log_schema.iteritems():
+        # When using special characters in the beginning or end
+        # of a column name, they have to be wrapped in backticks
+        key_name = '`{}`'.format(key_name)
+
+        special_key = None
+        # Transform the {} or [] into hashable types
+        if key_type == {}:
+            special_key = dict
+        elif key_type == []:
+            special_key = list
+        # Cast nested dict as a string for now
+        # TODO(jacknagz): support recursive schemas
+        elif isinstance(key_type, dict):
+            special_key = 'string'
+
+        # Account for envelope keys
+        if root_key:
+            if special_key is not None:
+                athena_schema[root_key][key_name] = SCHEMA_TYPE_MAPPING[special_key]
+            else:
+                athena_schema[root_key][key_name] = SCHEMA_TYPE_MAPPING[key_type]
+        else:
+            if special_key is not None:
+                athena_schema[key_name] = SCHEMA_TYPE_MAPPING[special_key]
+            else:
+                athena_schema[key_name] = SCHEMA_TYPE_MAPPING[key_type]
+
+
+def create_table(athena_client, options, config):
+    """Create a 'streamalert' Athena table
+
+    Args:
+        athena_client (boto3.client): Instantiated CLI AthenaClient
+        options (namedtuple): The parsed args passed from the CLI
+    """
+    if not options.bucket:
+        LOGGER_CLI.error('Missing command line argument --bucket')
+        return
+
+    if not options.refresh_type:
+        LOGGER_CLI.error('Missing command line argument --refresh_type')
+        return
+
+    if options.type == 'data':
+        if not options.table_name:
+            LOGGER_CLI.error('Missing command line argument --table_name')
+            return
+
+        if options.table_name not in terraform_cli_helpers.enabled_firehose_logs(config):
+            LOGGER_CLI.error('Table name %s missing from configuration or '
+                             'is not enabled.', options.table_name)
+            return
+
+        if athena_client.check_table_exists(options.table_name):
+            LOGGER_CLI.info('The \'%s\' table already exists.', options.table_name)
+            return
+
+        log_info = config['logs'][options.table_name.replace('_', ':', 1)]
+        schema = dict(log_info['schema'])
```

Comment:
isn't this already a dict? why is the casting necessary here?

Reply:
OrderedDict to dict